src/argilla/client/sdk/datasets/v1/models.py

```diff
@@ -0,0 +1,34 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from datetime import datetime
+from typing import Any, Dict
+
+from pydantic import BaseModel, Field
+
+
+class BaseDatasetModel(BaseModel):
+    tags: Dict[str, str] = Field(default_factory=dict)
+    metadata: Dict[str, Any] = Field(default_factory=dict)
+    name: str = Field(regex="^(?!-|_)[a-zA-Z0-9-_ ]+$")
```

Comment:
@jfcalvo I saw that datasets can be created using spaces, which was not supported before, is this correct?

Reply:
That's right. We don't have any limitation about what a dataset name can be. We will have soon a limitation in the number of characters but that's all.

Reply:
With that said maybe you don't need to specify any regex. That should be transparent to you.

Reply:
Since I was validating the schema of the dataset uploaded to Argilla, I changed that since the dataset I was using was named `"Big Dataset"` and the validation was failing, but we can make it just a string with no regex validation.